Source/WebKit/WebKit.xcodeproj/project.pbxproj

```diff
@@ -16212,6 +16213,27 @@
 			shellPath = /bin/sh;
 			shellScript = "# Touch Info.plist to let Xcode know it needs to copy it into the built product\nif [[ \"${CONFIGURATION}\" != \"Production\" ]]; then\ntouch \"${SRCROOT}/Info.plist\";\nfi;\n";
 		};
+		1C628FFF28EC99E700C26B60 /* Generate Private Umbrella Header */ = {
+			isa = PBXShellScriptBuildPhase;
+			buildActionMask = 2147483647;
+			files = (
+			);
+			inputFileListPaths = (
+			);
+			inputPaths = (
+				"$(SRCROOT)/Scripts/generate-umbrella-headers.py",
+				"$(SRCROOT)/Shared/API/Cocoa/WebKitPrivate.h",
```

Comment:
With only these two inputs, I don't think the script will rerun when new private headers are added/removed. You probably need to depend on something like `$(BUILT_PRODUCTS_DIR)/$(PRIVATE_HEADERS_FOLDER_PATH)`, and then set `USE_RECURSIVE_SCRIPT_INPUTS_IN_SCRIPT_PHASES=YES`.

This in itself is kind of an anti-pattern (that build setting is rarely used, and can make it easy to accidentally introduce dependency cycles).

Another hacky idea would be to add the headermap Xcode generates as an input, since that will be regenerated when headers are added/removed, but not every build.